device/mellanox/x86_64-mlnx_msn2700-r0/plugins/thermalutil.py

```diff
@@ -0,0 +1,456 @@
+            up to nearest thousandth of one degree Celsius, e.g. 30.125
+        """
+        if self.high_threshold is None:
+            return None
+        value_str = self._read_generic_file(self.high_threshold, 0)
+        if value_str is None:
+            return None
+        value_float = float(value_str)
+        if self.category == THERMAL_DEV_CATEGORY_MODULE and value_float == THERMAL_API_INVALID_HIGH_THRESHOLD:
+            return None
+        return value_float / 1000.0
+
+    def get_high_critical_threshold(self):
+        """
+        Retrieves the high critical threshold temperature of thermal
+
+        Returns:
+            A float number, the high critical threshold temperature of thermal in Celsius
+            up to nearest thousandth of one degree Celsius, e.g. 30.125
+        """
+        if self.high_critical_threshold is None:
+            return None
+        value_str = self._read_generic_file(self.high_critical_threshold, 0)
+        if value_str is None:
+            return None
+        value_float = float(value_str)
+        if self.category == THERMAL_DEV_CATEGORY_MODULE and value_float == THERMAL_API_INVALID_HIGH_THRESHOLD:
+            return None
+        return value_float / 1000.0
+
+
+class ThermalUtil(ThermalBase):
+    """Platform-specific Thermalutil class"""
+
+    MAX_PSU_FAN = 1
+    MAX_NUM_PSU = 2
+    GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
+    number_of_thermals = 0
+    thermal_list = []
+
+    def _get_sku_name(self):
+        p = subprocess.Popen(self.GET_HWSKU_CMD, shell=True, stdout=subprocess.PIPE)
+        out, err = p.communicate()
+        return out.rstrip('\n')
+
+    def __init__(self):
+        sku = self._get_sku_name()
+        # create thermal objects for all categories of sensors
+        tp_index = hwsku_dict_thermal[sku]
+        thermal_profile = thermal_profile_list[tp_index]
+        for category in thermal_device_categories_all:
+            if category == THERMAL_DEV_CATEGORY_AMBIENT:
+                count, ambient_list = thermal_profile[category]
+                for ambient in ambient_list:
+                    thermal = Thermal(category, ambient, True)
+                    self.thermal_list.append(thermal)
+            else:
+                start, count = 0, 0
+                if category in thermal_profile:
+                    start, count = thermal_profile[category]
+                    if count == 0:
+                        continue
+                if count == 1:
+                    thermal = Thermal(category, 0, False)
+                    self.thermal_list.append(thermal)
+                else:
+                    for index in range(count):
+                        thermal = Thermal(category, start + index, True)
+                        self.thermal_list.append(thermal)
+        self.number_of_thermals = len(self.thermal_list)
+
+    def get_num_thermals(self):
```

Comment:
get_num_thermal_objects?

Reply:
looks like the name get_num_xxx is the name convention used across plugins, like fan and PSU.